rw_test.go

```diff
@@ -0,0 +1,121 @@
+package srtgo
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+type tester struct {
+	blocking string          // blocking state
+	ctx      context.Context // done notification for senders
+	cancel   func()          // cancel tester
+	err      chan error      // error output for senders/receivers
+	started  sync.WaitGroup  // sync goroutines on start
+}
+
+// Creates a connector socket and keeps writing to it
+func (t *tester) send(port uint16) {
+	buf := make([]byte, 1316)
+	sock := NewSrtSocket("127.0.0.1", port, map[string]string{"blocking": t.blocking, "mode": "caller", "transtype": "file"})
+	if sock == nil {
+		t.err <- errors.New("failed to create caller socket")
+		return
+	}
+	defer sock.Close()
+
+	// Wait until all sockets are ready
+	t.started.Done()
+	t.started.Wait()
+
+	err := sock.Connect()
+	if err != nil {
+		t.err <- errors.Wrap(err, "connect")
```

Comment:
Wrapping of errors is also possible via fmt.Errorf (available since go 1.13), that way you wouldn't need to import the pkg/errors pkg

Reply:
ah good idea